addons/survey/wizard/survey_email_compose_message.py

```diff
@@ -140,24 +127,24 @@ def create_response_and_send_mail(wizard, token, partner_id, email):
                 'email_from': wizard.email_from or None,
                 'email_to': email,
             }
-            mail_id = mail_mail_obj.create(cr, uid, values, context=context)
-            mail_mail_obj.send(cr, uid, [mail_id], context=context)
+            mail = Mail.create(values)
+            mail.send()
 
         def create_token(wizard, partner_id, email):
-            if context.get("survey_resent_token"):
-                response_ids = survey_response_obj.search(cr, uid, [('survey_id', '=', wizard.survey_id.id), ('state', 'in', ['new', 'skip']), '|', ('partner_id', '=', partner_id), ('email', '=', email)], context=context)
-                if response_ids:
-                    return survey_response_obj.read(cr, uid, response_ids, ['token'], context=context)[0]['token']
+            if self.env.context.get("survey_resent_token"):
+                user_input = UserInput.search([('survey_id', '=', wizard.survey_id.id), ('state', 'in', ['new', 'skip']), '|', ('partner_id', '=', partner_id), ('email', '=', email)])
+                if user_input:
+                    return user_input.read(['token'])[0]['token']
             if wizard.public != 'email_private':
                 return None
             else:
                 token = uuid.uuid4().__str__()
                 # create response with token
-                survey_response_obj.create(cr, uid, {
+                UserInput.create({
                     'survey_id': wizard.survey_id.id,
                     'deadline': wizard.date_deadline,
                     'date_create': datetime.now(),
-                    'type': 'link',
+                    'input_type': 'link',
```

Comment:
Use odoo fields Date not python :telescope: 
```
'date_create': fields.Date.from_string(fields.Date.today())
````
and if use this then remove 
```
from datetime import datetime
```
no need more :crossed_flags: